src/line_cursor.ts

```diff
@@ -175,6 +175,27 @@ export class LineCursor extends Marker {
     return newNode;
   }
 
+  /**
+   * Returns true iff the node to which we would navigate if in() were
+   * called, which will be a validInLineNode, is also a validLineNode
+   * - in effect, if the LineCursor is at the end of the 'current
+   * line' of the program.
+   *
+   * This is used to determine whether to display an "Edit Block
+   * contents (→︎)" entry in the block context menu, to help users
+   * discover that the right arrow key can be used to navigate to
+   * block contents.
+   */
+  public atEndOfLine(): boolean {
+    const curNode = this.getCurNode();
+    if (!curNode) return false;
+    const rightNode = this.getNextNode(
+      curNode,
+      this.validInLineNode.bind(this),
+    );
+    return this.validLineNode(rightNode);
```

Comment:
Is this needed? `getNextNode` seems guaranteed to either return `null` or a 'valid' node (i.e. one for which `validInLineNode()` already returns `true`).

Reply:
Yes, because `getNextNode` returns nodes that are `validInLineNode`s, but only some of those are `validLineNode`s.  In particular, a field is a valid inline node but not a valid line node.

(This is actually the key feature of the whole PR!)

Reply:
It turns out that validLineNode and validInLineNode read the exact same to me without much more careful watching. That was the source of my confusion, actually. Thanks for clarifying.